### DISCUSSION TITLE
Do we need neighbors in node vlans?

### DISCUSSION THREAD

**Discussion by ipspace** [ipspace/netlab#887] [Features]
As I started working in a fix for #885, I figured out the existing VLAN code inconsistently creates the neighbors in node VLANs -- sometimes they are there, sometimes they are not.

I could make it work either way, but we have to decide which way we want to go. I would prefer not to have the neighbors in the **vlans** dictionary as they make no sense for routed VLANs, and we have neighbors on interfaces (VLAN interfaces, physical interface, or routed subinterfaces) anyway.

Does anyone see a good reason to keep **neighbors** list _in the per-node vlans data structure_ @jbemmel @ssasso @petercrocker

**Comment by jbemmel** [comment]
None of the existing templates seem to use it, most only use `bgp.neighbors` and one or two use `interface.neighbors` (e.g. mpls/eos.6pe.j2)

If we remove them and all the tests still pass, I'd say remove them

**Comment by ipspace** [comment]
@jbemmel: Thanks a million, I'll go with "no neighbors in VLAN data"

Obviously half of the tests fail regardless of which way I go, so I'll rebuild and manually check the test cases.
